plugins/synology-ss/src/main.ts

```diff
@@ -0,0 +1,283 @@
+                const d: Device = {
+                    providerNativeId: this.nativeId,
+                    name: camera.newName,
+                    nativeId: '' + camera.id,
+                    info: {
+                        manufacturer: camera.vendor,
+                        model: camera.model,
+                        firmware: camera.firmware,
+                        serialNumber: '' + camera.id
+                    },
+                    interfaces: [
+                        ScryptedInterface.Settings,
+                        ScryptedInterface.Camera,
+                        ScryptedInterface.VideoCamera,
+                    ],
+                    type: ScryptedDeviceType.Camera
+                };
+
+                devices.push(d);
+            }
+
+            for (const d of devices) {
+                await deviceManager.onDeviceDiscovered(d);
+            }
+
+            // todo: this was done, october 31st. remove sometime later.
+            // todo: uncomment after implementing per providerNativeId onDevicesChanged.
+            // await deviceManager.onDevicesChanged({
+            //     providerNativeId: this.nativeId,
+            //     devices
+            // });
+
+            for (const device of devices) {
+                this.getDevice(device.nativeId);
+            }
+        }
+        catch (e) {
+            this.log.a(`login error: ${e}`);
+            this.console.error('login error', e);
+        }
+    }
+
+    async getDevice(nativeId: string): Promise<any> {
+        await this.startup;
+        if (this.cameraDevices.has(nativeId))
+            return this.cameraDevices.get(nativeId);
+        const camera = this.cameras.find(camera => ('' + camera.id) === nativeId);
+        if (!camera)
+            throw new Error('camera not found?');
+        const ret = new SynologyCameraDevice(this, nativeId, camera);
+        this.cameraDevices.set(nativeId, ret);
+        return ret;
+    }
+
+    getSetting(key: string): string {
+        return this.storage.getItem(key);
+    }
+
+    async getSettings(): Promise<Setting[]> {
+        const ret: Setting[] = [
+            {
+                key: 'username',
+                title: 'Username',
+                value: this.getSetting('username'),
+            },
+            {
+                key: 'password',
+                title: 'Password',
+                type: 'password',
+                value: this.getSetting('password'),
+            },
+            {
+                key: 'url',
+                title: 'Synology Surveillance Station URL',
+                placeholder: 'http://192.168.1.100:5000',
+                value: this.getSetting('url'),
+            },
+        ];
+
+        return ret;
+    }
+
+    async putSetting(key: string, value: string | number) {
+        this.storage.setItem(key, value.toString());
+        this.discoverDevices(0);
+    }
+}
+
+export default new SynologySurveillanceStation();
```

Comment:
Unifi was using something like:

```ts
export default createInstanceableProviderPlugin("Unifi Protect Application", nativeid => new UnifiProtect(nativeid));
```

But I think you had issues with that. Was that resolved? It allows creation of multiple syno stations (optionally).

Reply:
Yeah, that was resolved. I removed it for simplicity at first, but could easily add it back in if you'd recommend?

Reply:
Unifi was the only plugin I saw using it, so wasn't sure if it was a recommended approach or a custom thing just for Unifi

Reply:
Yeah, it came up for Unifi, but I could/should add this to any "hub" based provider like Hue, etc.